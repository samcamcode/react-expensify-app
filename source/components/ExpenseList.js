import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses';

export const ExpenseList = (props) => (
    <div>
        {
            props.expenses.length === 0 ?
            (
                <p>No expenses</p>
            ) 
            :
            (
                props.expenses.map((expense) => {
                    return <ExpenseListItem 
                    id={expense.id} 
                    description={expense.description} 
                    amount={expense.amount} 
                    createdAt={expense.createdAt}
                    note={expense.note} />})
            )
        }
    </div>
);

const mapStateToProps = (state) =>{
    return {
        expenses: selectExpenses(state.expenses, state.filters )
    }
};

export default connect(mapStateToProps)(ExpenseList);






// HOC using a const ConnectedExpenseList not export default
// const ConnectedExpenseList = connect((state) =>{
//     return {
//         expenses: state.expenses
//     }
// })(ExpenseList);

// export default ConnectedExpenseList;