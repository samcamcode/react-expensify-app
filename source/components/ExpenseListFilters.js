import React from 'react';
import {connect} from 'react-redux';
import { setTextFilter, sortByAmount, sortByDate, setEndDate, setStartDate } from '../actions/filters';
import {DateRangePicker} from 'react-dates';

export class ExpenseListFilters extends React.Component {
    state = {
        calendarFocused: null
    };
    onDatesChanged = ({startDate, endDate}) => {
        this.props.setStartDate(startDate);
        this.props.setEndDate(endDate);
    };
    onFocusChange = (calendarFocused) => {
        this.setState(() => ({calendarFocused}));
    };
    onTextChange = (e) =>{
        this.props.setTextFilter(e.target.value);
    };
    onSortChange = (e) => {
        e.target.value === 'date' ? 
        this.props.sortByDate() : 
        this.props.sortByAmount();
    };

    render() {
        return(
            <div className="content-container">
                <div className="input-group">
                    <div className="input-group__item">
                        <input type='text' 
                        className="text-input"
                        placeholder="Search Expenses"
                        value={this.props.filters.text} 
                        onChange={this.onTextChange}/>
                    </div>
                    <div className="input-group__item">
                        <select 
                        className="select"
                        value={this.props.filters.sortBy} 
                        onChange={this.onSortChange}>
                        <option value='date'>Date</option>
                        <option value='amount'>Amount</option>
                        </select>
                    </div>
                    <div className="input-group__item">
                        <DateRangePicker 
                        startDate={this.props.filters.startDate}
                        endDate={this.props.filters.endDate}
                        onDatesChange={this.onDatesChanged}
                        focusedInput={this.state.calendarFocused}
                        onFocusChange={this.onFocusChange}
                        numberOfMonths={1}
                        isOutsideRange={() => false}
                        showClearDates={true}
                        />
                    </div>
                </div>
            </div>
        )
    }
};

const mapStateToProps = (state) => {
    return {
        filters: state.filters
    };
};

const mapDispatchToProps = (dispatch) => {
    return {
        setTextFilter: (text) => dispatch(setTextFilter(text)),
        sortByDate: () => dispatch(sortByDate()),
        sortByAmount: () => dispatch(sortByAmount()),
        setStartDate: (startDate) => dispatch(startDate(startDate)),
        setEndDate: (endDate) => dispatch(endDate(endDate))
    }
};

export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters);