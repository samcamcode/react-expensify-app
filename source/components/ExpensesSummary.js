import React from 'react';
import {connect} from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import totalExpenses from '../selectors/expenses-total';

export const ExpensesSummary = ({expensesCount, expensesTotal}) => {
    const expenseWord = expensesCount === 1 ? 'expense' : 'expsenses';
    const expensesFormattedTotal = numeral(expensesTotal / 100).format('$0,0.00');
    return (
        <div>
            <h3>viewing {expensesCount} {expenseWord} totaling {expensesFormattedTotal}</h3>
        </div>
    )
};

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);

    return {
        expensesCount: visibleExpenses.length,
        expensesTotal: totalExpenses(visibleExpenses),
    }
}

export default connect(mapStateToProps)(ExpensesSummary)