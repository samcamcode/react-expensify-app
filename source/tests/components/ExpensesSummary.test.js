import React from 'react';
import {shallow} from 'enzyme';
import {ExpensesSummary} from '../../components/ExpensesSummary';

test('should correctly render expenses summary with one expense', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={1} expensesTotal={235} />)
    expect(wrapper).toMatchSnapshot();
});

test('should correctly render expenses summary with multiple expense', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={10} expensesTotal={23500} />)
    expect(wrapper).toMatchSnapshot();
});