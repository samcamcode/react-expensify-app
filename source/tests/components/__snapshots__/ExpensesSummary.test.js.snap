// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render expenses summary with multiple expense 1`] = `
<div>
  <h3>
    viewing 
    10
     
    expsenses
     totaling 
    $235.00
  </h3>
</div>
`;

exports[`should correctly render expenses summary with one expense 1`] = `
<div>
  <h3>
    viewing 
    1
     
    expense
     totaling 
    $2.35
  </h3>
</div>
`;
